File: php/src/PWS_Json_Objects/Transaction.php
<?php
/**
 * Created by PhpStorm.
 * User: N13811A
 * Date: 9/4/14
 * Time: 12:11 PM
 */

namespace PWS_Json_Objects;


class Transaction implements \JsonSerializable{

    private $TransactionID;
    private $TransactionAmount;
    private $MarketCode;
    private $TransactionTimestamp;
    private $ClerkNumber;
    // Optional-Conditional
    private $CancelType;
    private $PaymentType;
    private $ReferenceNumber;
    private $DraftLocatorId;
    private $AuthorizationCode;
    private $OriginalAuthorizedAmount;
    private $CaptureAmount;
    private $CashBackAmount;
    private $OriginalTransactionTimestamp;
    private $OriginalSystemTraceId;
    private $OriginalSequenceNumber;
    private $OriginalAuthCode;
    private $NetworkResponseCode;
    private $ReversalReason;
    private $ReplacementAmount;
    private $OriginalReferenceNumber;
    private $TipAmount;
    private $ConvenienceFee;
    private $TaxExempt;
    private $Taxable;
    private $TaxAmount;
    private $PurchaseOrder;
    private $TokenRequested;
    private $SystemTraceID;

    public function JsonSerialize()
    {
        $vars = get_object_vars($this);

        return $vars;
    }

    /**
     * @param mixed $SystemTraceId
     */
    public function setSystemTraceID($SystemTraceId)
    {
        $this->SystemTraceID = $SystemTraceId;
    }

    /**
     * @return mixed
     */
    public function getSystemTraceID()
    {
        return $this->SystemTraceID;
    }



    /**
     * @param mixed $AuthorizationCode
     */
    public function setAuthorizationCode($AuthorizationCode)
    {
        $this->AuthorizationCode = $AuthorizationCode;
    }

    /**
     * @return mixed
     */
    public function getAuthorizationCode()
    {
        return $this->AuthorizationCode;
    }

    /**
     * @param mixed $CancelType
     */
    public function setCancelType($CancelType)
    {
        $this->CancelType = $CancelType;
    }

    /**
     * @return mixed
     */
    public function getCancelType()
    {
        return $this->CancelType;
    }

    /**
     * @param mixed $CaptureAmount
     */
    public function setCaptureAmount($CaptureAmount)
    {
        $this->CaptureAmount = $CaptureAmount;
    }

    /**
     * @return mixed
     */
    public function getCaptureAmount()
    {
        return $this->CaptureAmount;
    }

    /**
     * @param mixed $CashBackAmount
     */
    public function setCashBackAmount($CashBackAmount)
    {
        $this->CashBackAmount = $CashBackAmount;
    }

    /**
     * @return mixed
     */
    public function getCashBackAmount()
    {
        return $this->CashBackAmount;
    }

    /**
     * @param mixed $ClerkNumber
     */
    public function setClerkNumber($ClerkNumber)
    {
        $this->ClerkNumber = $ClerkNumber;
    }

    /**
     * @return mixed
     */
    public function getClerkNumber()
    {
        return $this->ClerkNumber;
    }

    /**
     * @param mixed $ConvenienceFee
     */
    public function setConvenienceFee($ConvenienceFee)
    {
        $this->ConvenienceFee = $ConvenienceFee;
    }

    /**
     * @return mixed
     */
    public function getConvenienceFee()
    {
        return $this->ConvenienceFee;
    }

    /**
     * @param mixed $DraftLocatorId
     */
    public function setDraftLocatorId($DraftLocatorId)
    {
        $this->DraftLocatorId = $DraftLocatorId;
    }

    /**
     * @return mixed
     */
    public function getDraftLocatorId()
    {
        return $this->DraftLocatorId;
    }

    /**
     * @param mixed $MarketCode
     */
    public function setMarketCode($MarketCode)
    {
        $this->MarketCode = $MarketCode;
    }

    /**
     * @return mixed
     */
    public function getMarketCode()
    {
        return $this->MarketCode;
    }

    /**
     * @param mixed $NetworkResponseCode
     */
    public function setNetworkResponseCode($NetworkResponseCode)
    {
        $this->NetworkResponseCode = $NetworkResponseCode;
    }

    /**
     * @return mixed
     */
    public function getNetworkResponseCode()
    {
        return $this->NetworkResponseCode;
    }

    /**
     * @param mixed $OriginalAmount
     */
    public function setOriginalAuthorizedAmount($OriginalAmount)
    {
        $this->OriginalAuthorizedAmount = $OriginalAmount;
    }

    /**
     * @return mixed
     */
    public function getOriginalAuthorizedAmount()
    {
        return $this->OriginalAuthorizedAmount;
    }

    /**
     * @param mixed $OriginalAuthCode
     */
    public function setOriginalAuthCode($OriginalAuthCode)
    {
        $this->OriginalAuthCode = $OriginalAuthCode;
    }

    /**
     * @return mixed
     */
    public function getOriginalAuthCode()
    {
        return $this->OriginalAuthCode;
    }

    /**
     * @param mixed $OriginalReferenceNumber
     */
    public function setOriginalReferenceNumber($OriginalReferenceNumber)
    {
        $this->OriginalReferenceNumber = $OriginalReferenceNumber;
    }

    /**
     * @return mixed
     */
    public function getOriginalReferenceNumber()
    {
        return $this->OriginalReferenceNumber;
    }

    /**
     * @param mixed $OriginalSequenceNumber
     */
    public function setOriginalSequenceNumber($OriginalSequenceNumber)
    {
        $this->OriginalSequenceNumber = $OriginalSequenceNumber;
    }

    /**
     * @return mixed
     */
    public function getOriginalSequenceNumber()
    {
        return $this->OriginalSequenceNumber;
    }

    /**
     * @param mixed $OriginalSystemTraceId
     */
    public function setOriginalSystemTraceId($OriginalSystemTraceId)
    {
        $this->OriginalSystemTraceId = $OriginalSystemTraceId;
    }

    /**
     * @return mixed
     */
    public function getOriginalSystemTraceId()
    {
        return $this->OriginalSystemTraceId;
    }

    /**
     * @param mixed $OriginalTransactionTimestamp
     */
    public function setOriginalTransactionTimestamp($OriginalTransactionTimestamp)
    {
        $this->OriginalTransactionTimestamp = $OriginalTransactionTimestamp;
    }

    /**
     * @return mixed
     */
    public function getOriginalTransactionTimestamp()
    {
        return $this->OriginalTransactionTimestamp;
    }

    /**
     * @param mixed $PaymentType
     */
    public function setPaymentType($PaymentType)
    {
        $this->PaymentType = $PaymentType;
    }

    /**
     * @return mixed
     */
    public function getPaymentType()
    {
        return $this->PaymentType;
    }

    /**
     * @param mixed $PurchaseOrder
     */
    public function setPurchaseOrder($PurchaseOrder)
    {
        $this->PurchaseOrder = $PurchaseOrder;
    }

    /**
     * @return mixed
     */
    public function getPurchaseOrder()
    {
        return $this->PurchaseOrder;
    }

    /**
     * @param mixed $ReferenceNumber
     */
    public function setReferenceNumber($ReferenceNumber)
    {
        $this->ReferenceNumber = $ReferenceNumber;
    }

    /**
     * @return mixed
     */
    public function getReferenceNumber()
    {
        return $this->ReferenceNumber;
    }

    /**
     * @param mixed $ReplacementAmount
     */
    public function setReplacementAmount($ReplacementAmount)
    {
        $this->ReplacementAmount = $ReplacementAmount;
    }

    /**
     * @return mixed
     */
    public function getReplacementAmount()
    {
        return $this->ReplacementAmount;
    }

    /**
     * @param mixed $ReversalReason
     */
    public function setReversalReason($ReversalReason)
    {
        $this->ReversalReason = $ReversalReason;
    }

    /**
     * @return mixed
     */
    public function getReversalReason()
    {
        return $this->ReversalReason;
    }

    /**
     * @param mixed $TaxAmount
     */
    public function setTaxAmount($TaxAmount)
    {
        $this->TaxAmount = $TaxAmount;
    }

    /**
     * @return mixed
     */
    public function getTaxAmount()
    {
        return $this->TaxAmount;
    }

    /**
     * @param mixed $TaxExempt
     */
    public function setTaxExempt($TaxExempt)
    {
        $this->TaxExempt = $TaxExempt;
    }

    /**
     * @return mixed
     */
    public function getTaxExempt()
    {
        return $this->TaxExempt;
    }

    /**
     * @param mixed $Taxable
     */
    public function setTaxable($Taxable)
    {
        $this->Taxable = $Taxable;
    }

    /**
     * @return mixed
     */
    public function getTaxable()
    {
        return $this->Taxable;
    }

    /**
     * @param mixed $TipAmount
     */
    public function setTipAmount($TipAmount)
    {
        $this->TipAmount = $TipAmount;
    }

    /**
     * @return mixed
     */
    public function getTipAmount()
    {
        return $this->TipAmount;
    }

    /**
     * @param mixed $TokenRequested
     */
    public function setTokenRequested($TokenRequested)
    {
        $this->TokenRequested = $TokenRequested;
    }

    /**
     * @return mixed
     */
    public function getTokenRequested()
    {
        return $this->TokenRequested;
    }

    /**
     * @param mixed $TransactionAmount
     */
    public function setTransactionAmount($TransactionAmount)
    {
        $this->TransactionAmount = $TransactionAmount;
    }

    /**
     * @return mixed
     */
    public function getTransactionAmount()
    {
        return $this->TransactionAmount;
    }

    /**
     * @param mixed $TransactionID
     */
    public function setTransactionID($TransactionID)
    {
        $this->TransactionID = $TransactionID;
    }

    /**
     * @return mixed
     */
    public function getTransactionID()
    {
        return $this->TransactionID;
    }

    /**
     * @param mixed $TransactionTimestamp
     */
    public function setTransactionTimestamp($TransactionTimestamp)
    {
        $this->TransactionTimestamp = $TransactionTimestamp;
    }

    /**
     * @return mixed
     */
    public function getTransactionTimestamp()
    {
        return $this->TransactionTimestamp;
    }


} 

File: python/src/apigee_objects/transaction.py
class Transaction(object):
        TransactionID = None
        TransactionAmount = None
        MarketCode = None
        TransactionTimestamp = None
        ClerkNumber = None
        CancelType = None
        PaymentType = None
        ReferenceNumber = None
        DraftLocatorId = None
        AuthorizationCode = None
        OriginalAuthorizedAmount = None
        CaptureAmount = None
        CashBackAmount = None
        OriginalTransactionTimestamp = None
        OriginalSystemTraceId = None
        OriginalSequenceNumber = None
        OriginalAuthCode = None
        NetworkResponseCode = None
        ReversalReason = None
        ReplacementAmount = None
        OriginalReferenceNumber = None
        TipAmount = None
        ConvenienceFee = None
        TaxExempt = None
        Taxable = None
        TaxAmount = None
        PurchaseOrder = None
        TokenRequested = None
        SystemTraceID = None


        def get_transaction(self):

            return_values = {}
            if not self.TransactionID:
                return_values['TransactionID': self.TransactionID]
                print 'true'
            if not self.TransactionAmount:
                return_values['TransactionAmount' : self.TransactionAmount]
            if not self.MarketCode:
                return_values['MarketCode' : self.MarketCode]
            if not self.TransactionTimestamp:
                return_values['TransactionTimestamp' : self.TransactionTimestamp]
            if not self.ClerkNumber:
                return_values['ClerkNumber' : self.ClerkNumber]
            if not self.CancelType:
                return_values['CancelType' : self.CancelType]
                print 'true'
            if not self.PaymentType: return_values['PaymentType' : self.PaymentType]
            if not self.ReferenceNumber: return_values['ReferenceNumber' : self.ReferenceNumber]
            if not self.DraftLocatorId: return_values['DraftLocatorId' : self.DraftLocatorId]
            if not self.AuthorizationCode: return_values['AuthorizationCode' : self.AuthorizationCode]
            if not self.OriginalAuthorizedAmount: return_values['OriginalAuthorizedAmount' : self.OriginalAuthorizedAmount]
            if not self.CaptureAmount: return_values['CaptureAmount' : self.CaptureAmount]
            if not self.CashBackAmount: return_values['CashBackAmount' : self.CashBackAmount]
            if not self.OriginalTransactionTimestamp: return_values['OriginalTransactionTimestamp' : self.OriginalTransactionTimestamp]
            if not self.OriginalSystemTraceId: return_values['OriginalSystemTraceId' : self.OriginalSystemTraceId]
            if not self.OriginalSequenceNumber: return_values['OriginalSequenceNumber' : self.OriginalSequenceNumber]
            if not self.OriginalAuthCode: return_values['OriginalAuthCode' : self.OriginalAuthCode]
            if not self.NetworkResponseCode: return_values['NetworkResponseCode' : self.NetworkResponseCode]
            if not self.ReversalReason: return_values['ReversalReason' : self.ReversalReason]
            if not self.ReplacementAmount: return_values['ReplacementAmount' : self.ReplacementAmount]
            if not self.OriginalReferenceNumber: return_values['OriginalReferenceNumber' : self.OriginalReferenceNumber]
            if not self.TipAmount: return_values['TipAmount' : self.TipAmount]
            if not self.ConvenienceFee: return_values['ConvenienceFee' : self.ConvenienceFee]
            if not self.TaxExempt: return_values['TaxExempt' : self.TaxExempt]
            if not self.PurchaseOrder: return_values['PurchaseOrder' : self.PurchaseOrder]
            if not self.TokenRequested: return_values['TokenRequested' : self.TokenRequested]
            if not self.SystemTraceID: return_values['SystemTraceID' : self.SystemTraceID]

            return return_values